### PR TITLE
Improved UI/UX for new routing, including bug fixes

### PR DESCRIPTION
**Pull request by ajb**
**this PR is awaiting a fix for the infinite loading of the FarmInfo component**

(This PR is dependent on my other PRs, #142 and #143.)

Tagging for review: @malytinKonstantin @cryptopower80 @papasnowee

As I was getting everything up and running locally, I noticed that the recent refactoring introduced some small/not-so-small bugs:

- In the dev environment, if you connected a wallet and went to your user dashboard, then hit "refresh" in your browser, the page would 404. I proposed a fix for this in #143, but this was actually masking a bigger bug:

- If you were on the user dashboard and refreshed the browser, you would still be on the dashboard page. But this required your wallet to already be connected, which didn't work because we had `cacheProvider: false` in the web3modal config.

- Additionally, the UX for viewing a non-wallet address was really weird. It actually required you to connect your wallet, *then* you entered the wallet address that you wanted to view! I took the liberty of refactoring this, since I even saw in the code that there was a `TODO` comment about persisting the address to localstorage. I decided to use url params instead, so the address for viewing a wallet address is `[dashboard url]/check-balance/0x000000`. This is better, I think.

The homepage now looks like this:

Screen Shot 2021-06-04 at 5 36 43 PM

I think it makes a lot more sense 😄 

I tested this and it seems to be working well, but there's definitely a chance that I could have missed something.

Thanks all!

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -2,21 +2,21 @@
...
     <meta name="description" content="Harvest.finance management tool" />
-    <link rel="apple-touch-icon" href="./favicon.ico" />
-    <link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@700&family=Roboto:wght@300;400&display=swap" rel="stylesheet"> 
+    <link rel="apple-touch-icon" href="/favicon.ico" />
+    <link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@700&family=Roboto:wght@300;400&display=swap" rel="stylesheet">
     <!-- GOOGLE FONTS -->
     <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous"><!--font Awesome-->
 
-    
+
```
Comment (ajb) [inline public/index.html:13]: Sorry, I have my editor set to remove trailing spaces.

File: src/components/farmInfo/FarmInfo.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const FarmInfo: React.FC<FarmInfoProps> = observer((props) => {
     savedGasStore.isFetching ||
     apyStore.isFetching
 
-  const displayApy = apy && apy !== '0' ? `${apy}%` : 'Error'
+  const displayApy = apy && apy !== '0' ? `${apy}%` : '-'
```
Comment (ajb) [inline src/components/farmInfo/FarmInfo.tsx:32]: This is used as a loading state, so we don't want to show the word "error"

File: src/App/App.tsx [hunk 3]
```diff
@@ -15,7 +15,13 @@ import { useStores } from '@/stores/utils'
 import './styles/App.scss'
 
 export const App = observer(() => {
-  const { settingsStore } = useStores()
+  const { settingsStore, apyStore, farmPriceStore } = useStores()
+
+  // This data can be fetched on load since it's not address-dependent
+  useEffect(() => {
+    apyStore.fetch()
+    farmPriceStore.fetch()
```
Comment (papasnowee) [inline src/App/App.tsx:23]: Why is it in the App file? if we add everything to the APP file, then the APP file will very soon turn into a trashcan

File: src/components/EnterReadOnlyAddress/EnterReadOnlyAddress.tsx [hunk 4]
```diff
@@ -5,26 +5,10 @@ import { ValidationMessage, Panel } from './styles'
...
-    return false
-  }
-
-  return true
-}
-
-type CheckBalanceProps = {}
-
-export const CheckBalance: React.FC<CheckBalanceProps> = observer((props) => {
+export const EnterReadOnlyAddress: React.FC = observer((props) => {
```
[the remark this answers is not in the document]
Comment (ajb) [reply src/components/EnterReadOnlyAddress/EnterReadOnlyAddress.tsx:11]: Sorry, I should have explained. 

We have a page called "CheckBalance", so I thought it was best to make this component have a different name for clarity. I think it is much better like this, but I can change it back if you would prefer.

File: src/pages/UserDashboard/UserDashboard.tsx [hunk 5]
```diff
@@ -1,31 +1,39 @@
...
+    }
+  }, [appStore.address]) // this effect runs whenever appStore.address is changed
 
   return (
     <Styled.Main>
       <>
         <Wallet address={appStore.address} />
         <FarmInfo
-          isLoadingAssets={assetsStore.isFetching}
+          isLoadingAssets={!assetsStore.isFetched}
```
[the remark this answers is not in the document]
Comment (ajb) [reply src/pages/UserDashboard/UserDashboard.tsx:36]: No problem, reverting!

**Comment by ajb** [comment]
@papasnowee thank you again for the comments. I think all are resolved :)

File: src/components/EnterReadOnlyAddress/EnterReadOnlyAddress.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (papasnowee) [reply src/components/EnterReadOnlyAddress/EnterReadOnlyAddress.tsx:11]: I offered to rename

File: src/stores/settings-store.ts [hunk 6]
```diff
@@ -27,12 +27,8 @@ class SettignsStore {
   }
 
   private async init() {
-    // TODO fix
-    // this.readCache()
-    // const exchangeRatesResponse = await exchangeRatesStore.fetch('?base=USD')
-
-    // this.settings.currency.options = Object.keys(exchangeRatesResponse)
-    this.inited = true
+    this.readCache()
```
Comment (papasnowee) [inline src/stores/settings-store.ts:30]: Deleted comments in vain, exchangeRatesStore still needs to be improved. It wiil need to switch currencies.

Comment (ajb) [reply src/stores/settings-store.ts:30]: Maybe I got my PRs mixed up. I have the price feeds ready to go but haven't submitted it. We can add the comments back for now.

Comment (papasnowee) [reply src/stores/settings-store.ts:30]: > Maybe I got my PRs mixed up. I have the price feeds ready to go but haven't submitted it. We can add the comments back for now.

I didn’t know that you have already solved this problem, then comments can be deleted

**Comment by ajb** [comment]
Should be good to go!

**Pull request closed without merging**